packages/sdk/src/types/internal/context.ts

```diff
@@ -190,6 +193,12 @@ export class InternalContext {
     }
 
     private createActorFromPayload(payload: CreateActorCommon): ForwardPromise<Actor> {
+        // bind all user group mappings before serialization
+        const ug = safeGet(payload, 'actor', 'appearance', 'enabled');

```

Comment:
This is brittle. It will be cleaner and more robust to pass context in the constructor.